framework/tlvf/src/src/tlvftypes.cpp

```diff
@@ -0,0 +1,81 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include <climits>
+#include <linux/if_ether.h>
+#include <tlvf/tlvftypes.h>
+
+namespace tlvf {
+std::string int_to_hex_string(const unsigned int integer, const uint8_t number_of_digits)
+{
+    // 'number_of_digits' represent how much digits the number should have, so the function will
+    // pad the number with zeroes from left, if necessary.
+    // for example: int_to_hex_string(255, 4) -> "00ff"
+    //              int_to_hex_string(255, 1) -> "ff"
+
+    std::stringstream ss_hex_string;
+
+    // convert to hex
+    ss_hex_string << std::setw(number_of_digits) << std::setfill('0') << std::hex << integer;
+
+    return ss_hex_string.str();
+};
+
+// Converts a mac address in a human-readable format
+std::string mac_to_string(const uint8_t *mac_address)
+{
+    std::string mac_addr_string;
+
+    mac_addr_string = int_to_hex_string((uint32_t)mac_address[0], 2) + ":" +
+                      int_to_hex_string((uint32_t)mac_address[1], 2) + ":" +
+                      int_to_hex_string((uint32_t)mac_address[2], 2) + ":" +
+                      int_to_hex_string((uint32_t)mac_address[3], 2) + ":" +
+                      int_to_hex_string((uint32_t)mac_address[4], 2) + ":" +
+                      int_to_hex_string((uint32_t)mac_address[5], 2);
```

Comment:
Write a loop instead of repeating 6 times the same line

Reply:
Its not the same line, the last one doesn't include the `":"` separator.

Reply:
I meant this:
```c
for (size_t i = 0; i < ETHA_LEN; i++) {
    if (i > 0) {
        mac_addr_string += mac_bytes_separator;
    }
    mac_addr_string += int_to_hex_string(mac_address[i], 2);
}